src/lib/utils/memory_usage.cpp

```diff
@@ -0,0 +1,35 @@
+#include "memory_usage.hpp"
+
+#include <iomanip>
+
+namespace opossum {
+
+MemoryUsage::MemoryUsage(const size_t bytes) : bytes(bytes) {}
+
+void MemoryUsage::print(std::ostream& stream) const {
```

Comment:
Why so complicated? Not completely tested, but this should work:

```
std::string to_byte_string(size_t bytes) {
    if (bytes < 1000) return std::to_string(bytes) + " B";

    const constexpr char *prefixes = "kMGT";

    auto log1000 = 0;
    float bytes_float = bytes;
    while(bytes_float >= 1000) {
    	++log1000;
    	bytes_float /= 1000;
    }

    std::stringstream out;
    out << std::fixed << std::setprecision(1) << bytes_float << " " << prefixes[log1000] << "B";

    return out.str();
}
```

Reply:
Nope, every python programmer can read my code, yours is the more complicated version. Looking over it twice I have just about a rough guess what it might do. I agree yours might "scale" better.

Reply:
But... but... just before you talked about billions of rows!